### PR TITLE
Fixes a Couple of Things Noticed While Reviewing Previous PR

### PR DESCRIPTION
**Pull request by schroedingerscode**
## PR Description
- Adds universal gson versioning, all using `gradle/versions.gradle` files
- Reverts Constants to spec defaults
- Removes ordering constraint on one parallel stream

### CODE REVIEW

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/ValidatorsUtil.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
     List<Integer> active_validator_indices = new ArrayList<>();
     IntStream.range(0, validators.size())
         .parallel()
-        .forEachOrdered(
+        .forEach(
```
Comment (cemozerr) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/ValidatorsUtil.java:66]: idk what the difference between forEach and forEachOrdered is but to me this seems like using parallel processing to add to the same list could create concurrency issues.

Comment (schroedingerscode) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/ValidatorsUtil.java:66]: So forEachOrdered guarantees the process order is the same as the encountered order. I'd imagine it has worse performance as compared to forEach when doing parallel stream processing. In this case, because all that we're doing is adding indices that meet a certain condition to a list, output order isn't important. 

tl;dr I think we're safe here. Good eyes though, these are exactly the type of things we need to keep an eye on.

File: data/adapter/build.gradle [hunk 2]
```diff
@@ -16,7 +16,7 @@ dependencies {
   implementation 'net.consensys.cava:cava-units'
   implementation 'net.consensys.cava:cava-ssz'
   implementation 'com.google.guava:guava'
-  implementation 'com.google.code.gson:gson:2.8.5'
```
Comment (jrhea) [inline data/adapter/build.gradle:19]: this has been bugging me.  good fix

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/ValidatorsUtil.java (hunk 1 above)
Comment (jrhea) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/ValidatorsUtil.java:66]: agreed on both counts.  good tweak

**Review by jrhea — APPROVED** [review]
lgtm
